File: SCTE65/src/java/other/org/himalay/si/CDS_Record.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
// 
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.si;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Sun Sep 14 22:48:36 EDT 2014")
public class CDS_Record extends BinStruct implements PublicBinMsg {

	// members variables
	// number_of_carriers
	public short number_of_carriers;
	// bf1
	public BitField_16 bf1;
	// bf2
	public BitField_16 bf2;

	public CDS_Record() // throws Exception
	{
		init();
	}

	private void init() {
		// Initialize number_of_carriers

		// Initialize bf1
		bf1 = new BitField_16();
		// Initialize bf2
		bf2 = new BitField_16();
	}

	public int readNoHeader(DataInputStream istream) throws IOException {

		return read(istream);
	}

	public int read(DataInputStream istream) throws IOException {
		preRead();
		int retVal = 0;

		// read number_of_carriers
		{
			number_of_carriers = (short) (istream.readUnsignedByte());
			retVal += 1;
		}
		// read bf1
		retVal += bf1.read(istream);
		// read bf2
		retVal += bf2.read(istream);

		postRead();
		return retVal;
	}

	public int write(DataOutputStream ostream) throws IOException {
		preWrite();
		int retVal = 0;

		// write number_of_carriers
		ostream.writeByte(number_of_carriers);
		retVal += 1;
		// write bf1
		ostream.writeShort(bf1.getValue());
		retVal += 2;
		// write bf2
		ostream.writeShort(bf2.getValue());
		retVal += 2;
		postWrite();
		return retVal;
	}

	public int dump(DumpContext dc) throws IOException {
		dc.indent();
		dc.getPs().print("CDS_Record\n");
		dc.increaseIndent();
		int retVal = 0;
		// write number_of_carriers
		dc.indent();
		dc.getPs().println(
				"number_of_carriers=" + number_of_carriers + "(0x"
						+ Integer.toHexString(number_of_carriers) + ")");
		// write bf1
		{
			dc.indent();
			dc.getPs().print("spacingUnit: ");
			dc.getPs().println(BitField_8.toDisplayString(getSpacingUnit(), 1));
			dc.indent();
			dc.getPs().print("frequencySpacing: ");
			dc.getPs().println(
					BitField_8.toDisplayString(getFrequencySpacing(), 14));
		}
		// write bf2
		{
			dc.indent();
			dc.getPs().print("frequencyUnit: ");
			dc.getPs().println(
					BitField_8.toDisplayString(getFrequencyUnit(), 1));
			dc.indent();
			dc.getPs().print("firstCarrierFrequency: ");
			dc.getPs().println(
					BitField_8.toDisplayString(getFirstCarrierFrequency(), 15));
		}
		dc.decreaseIndent();
		return retVal;
	}

	// Getter for number_of_carriers
	// public short getNumber_of_carriers()
	// {
	// return number_of_carriers ;
	// }

	// Setter for number_of_carriers
	// public void setNumber_of_carriers(short val)
	// {
	// this.number_of_carriers= val;
	// }
	// Getter for bf1
	// public BitField_16 getBf1()
	// {
	// return bf1 ;
	// }

	// Setter for bf1
	// public void setBf1(BitField_16 val)
	// {
	// this.bf1= val;
	// }
	// Getter for bf2
	// public BitField_16 getBf2()
	// {
	// return bf2 ;
	// }

	// Setter for bf2
	// public void setBf2(BitField_16 val)
	// {
	// this.bf2= val;
	// }

	public int getSpacingUnit() {
		return (bf1.getValue() & 0x00008000) >> 15;
	}

	public void setSpacingUnit(int val) {
		bf1.setValue((bf1.getValue() & 0xffff7fff) | ((val << 15) & 0x00008000));
	}

	public int getFrequencySpacing() {
		return (bf1.getValue() & 0x00003fff) >> 0;
	}

	public void setFrequencySpacing(int val) {
		bf1.setValue((bf1.getValue() & 0xffffc000) | ((val << 0) & 0x00003fff));
	}

	public int getFrequencyUnit() {
		return (bf2.getValue() & 0x00008000) >> 15;
	}

	public void setFrequencyUnit(int val) {
		bf2.setValue((bf2.getValue() & 0xffff7fff) | ((val << 15) & 0x00008000));
	}

	public int getFirstCarrierFrequency() {
		return (bf2.getValue() & 0x00007fff) >> 0;
	}

	public void setFirstCarrierFrequency(int val) {
		bf2.setValue((bf2.getValue() & 0xffff8000) | ((val << 0) & 0x00007fff));
	}

	public int getSize() throws IOException {
		DataOutputStream dos = new DataOutputStream(new NullStream());
		return this.write(dos);
	}

}

// End of code

File: libsi/src/bp/CDS_Record.cpp
//**//
//@Created(date = "Mon Feb 01 22:29:58 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "CDS_Record.h"

#ifdef org_himalay_si_CDS_Record__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace si {

using namespace ::org::himalay::msgs::runtime;

// Constructor
CDS_Record::CDS_Record() {
    references= 0;


    // number_of_carriers
    number_of_carriers= (ui8)0;
    // bf1
    bf1 = 0;
    // bf2
    bf2 = 0;
}
// Destructor
CDS_Record::~CDS_Record() {
    // number_of_carriers

    // bf1

    // bf2

}

int CDS_Record::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // number_of_carriers
    {
        number_of_carriers=(istream.readUI8());
        retVal+=1;
    }
    // bf1
    bf1= istream.readUI16();
    retVal += 2;
    // bf2
    bf2= istream.readUI16();
    retVal += 2;
    return retVal;
}

int CDS_Record::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read number_of_carriers
    {
        number_of_carriers=(istream.readUI8());
        retVal+=1;
    }
    // read bf1
    bf1= istream.readUI16();
    retVal += 2;
    // read bf2
    bf2= istream.readUI16();
    retVal += 2;

    return retVal;
}

int CDS_Record::write(DataOutputStream& ostream) { // throws IOException




    int retVal= 0;
    // write number_of_carriers
    ostream.writeUI8(number_of_carriers);
    retVal +=1;
    // write bf1
    ostream.writeUI16(bf1);
    retVal +=2;
    // write bf2
    ostream.writeUI16(bf2);
    retVal +=2;

    return retVal;
}

int CDS_Record::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("CDS_Record\n");
    dc.increaseIndent();
    int retVal= 0;
    // write number_of_carriers
    dc.indent();
    dc.getPs().print("number_of_carriers=");
    dc.getPs().println((long)number_of_carriers);
    // write bf1
    {
        dc.indent();
        dc.getPs().print("spacingUnit: ");
        dc.getPs().printlnBin(getSpacingUnit(),1);
        dc.indent();
        dc.getPs().print("frequencySpacing: ");
        dc.getPs().printlnBin(getFrequencySpacing(),14);
    }
    // write bf2
    {
        dc.indent();
        dc.getPs().print("frequencyUnit: ");
        dc.getPs().printlnBin(getFrequencyUnit(),1);
        dc.indent();
        dc.getPs().print("firstCarrierFrequency: ");
        dc.getPs().printlnBin(getFirstCarrierFrequency(),15);
    }
    dc.decreaseIndent();
    return retVal;
}


// Getter for number_of_carriers
//ui8 CDS_Record::getNumber_of_carriers()
//{
//return this->number_of_carriers;
//}

// Setter for number_of_carriers
//void CDS_Record::setNumber_of_carriers(ui8 val)
//{
//this->number_of_carriers= val;
//}

// Getter for bf1
//ui16 CDS_Record::getBf1()
//{
//return this->bf1;
//}

// Setter for bf1
//void CDS_Record::setBf1(ui16 val)
//{
//this->bf1= val;
//}

// Getter for bf2
//ui16 CDS_Record::getBf2()
//{
//return this->bf2;
//}

// Setter for bf2
//void CDS_Record::setBf2(ui16 val)
//{
//this->bf2= val;
//}

int CDS_Record::getSpacingUnit( ) {
    return ( this->bf1 & 0x00008000 ) >> 15 ;
}

void CDS_Record::setSpacingUnit(int val) {
    bf1=(  (this->bf1 & 0xffff7fff) | ( (val << 15 ) & 0x00008000));
}

int CDS_Record::getFrequencySpacing( ) {
    return ( this->bf1 & 0x00003fff ) >> 0 ;
}

void CDS_Record::setFrequencySpacing(int val) {
    bf1=(  (this->bf1 & 0xffffc000) | ( (val << 0 ) & 0x00003fff));
}

int CDS_Record::getFrequencyUnit( ) {
    return ( this->bf2 & 0x00008000 ) >> 15 ;
}

void CDS_Record::setFrequencyUnit(int val) {
    bf2=(  (this->bf2 & 0xffff7fff) | ( (val << 15 ) & 0x00008000));
}

int CDS_Record::getFirstCarrierFrequency( ) {
    return ( this->bf2 & 0x00007fff ) >> 0 ;
}

void CDS_Record::setFirstCarrierFrequency(int val) {
    bf2=(  (this->bf2 & 0xffff8000) | ( (val << 0 ) & 0x00007fff));
}

int CDS_Record::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_si_CDS_Record__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::si::CDS_Record* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::si::CDS_Record* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code